symfit/core/minimizers.py

```diff
@@ -31,11 +32,36 @@ def __init__(self, objective, parameters):
         """
         self.parameters = parameters
         self._fixed_params = [p for p in parameters if p.fixed]
-        self.objective = partial(objective, **{p.name: p.value for p in self._fixed_params})
+        self.objective = self._baseobjective_from_callable(objective)
```

Comment:
My more extensive comments elsewhere also means that this should be moved to `Fit`.